=== modified file 'openerp/addons/base/ir/ir_model.py'
--- openerp/addons/base/ir/ir_model.py  2012-02-27 10:09:53 +0000
+++ openerp/addons/base/ir/ir_model.py  2012-06-13 15:20:55 +0000
@@ -692,12 +692,12 @@
         action_id = False
 
         if xml_id:
-            cr.execute('''SELECT imd.id, imd.res_id, md.id, imd.model
+            cr.execute('''SELECT imd.id, imd.res_id, md.id, imd.model, imd.noupdate
                           FROM ir_model_data imd LEFT JOIN %s md ON (imd.res_id = md.id)
                           WHERE imd.module=%%s AND imd.name=%%s''' % model_obj._table,
                           (module, xml_id))
             results = cr.fetchall()
-            for imd_id2,res_id2,real_id2,real_model in results:
+            for imd_id2,res_id2,real_id2,real_model,noupdate2 in results:
                 if not real_id2:
                     self._get_id.clear_cache(self, uid, module, xml_id)
                     self.get_object_reference.clear_cache(self, uid, module, xml_id)
@@ -707,6 +707,9 @@
                     assert model == real_model, "External ID conflict, %s already refers to a `%s` record,"\
                         " you can't define a `%s` record with this ID." % (xml_id, real_model, model)
                     res_id,action_id = res_id2,imd_id2
+                    
+                if noupdate2:
+                    return res_id
 
         if action_id and res_id:
             model_obj.write(cr, uid, [res_id], values, context=context)
